src/main/kotlin/com/demonwav/mcdev/platform/mixin/inspection/MixinClassReferenceInspection.kt

```diff
@@ -41,7 +41,7 @@ class MixinClassReferenceInspection : MixinInspection() {
                 return
             }
 
-            holder.registerProblem(type, "${referencedClass.name} is a Mixin class, and cannot be referenced directly")
+            holder.registerProblem(type, "Mixin class cannot be referenced directly")
```

Comment:
I didn't add this inspection (I've just ported it to Kotlin), but one case that was broken was if referenced Mixin classes were either the Mixin itself or one of its super mixins (https://github.com/minecraft-dev/MinecraftDev/issues/102). I've only fixed that case currently. I don't know much about accessor Mixins so this is probably not handled currently.